app/assets/javascripts/discourse/tests/acceptance/sidebar-test.js

```diff
@@ -63,3 +69,210 @@ acceptance("Sidebar - User with sidebar enabled", function (needs) {
     assert.ok(exists(".sidebar-container"), "displays the sidebar");
   });
 });
+acceptance("Sidebar - section API", function (needs) {
```

Comment:
This file is a getting a little crowded. I'll suggest splitting this into `sidebar-plugin-api-test` or something.